client/scripts/controllers/server/socket/index.ts

```diff
@@ -1,96 +1,83 @@
-import { IWebsocketsPayload, WebsocketMessageType } from 'types';
+import { io } from 'socket.io-client';
+import { ChainEventsNamespace } from 'controllers/server/socket/chainEventsNs';
 
-// how long to wait after being disconnected
-export const RECONNECT_DELAY = 5000;
-// how often to send heartbeats to avoid being disconnected
-export const HEARTBEAT_DELAY = 15000;
+export class WebSocketController {
+  private _socket;
+  private _isConnected = false;
+  public readonly chainEventsNs: ChainEventsNamespace;
```

Comment:
Are the ChainEventNamespace API messages documented anywhere? Would be nice to have a reference describing what it emits/receives. We could also document them alongside creating typescript for the events, ideally in `shared/` so it can reused against both client/server.

Reply:
added a readme in `server/socket` with a short description of each custom event. Also added `ChainEventNotification` type in `shared/types`.